pkg/storage/stores/shipper/indexgateway/config.go

```diff
@@ -47,6 +47,26 @@ const (
 	RingMode Mode = "ring"
 )
 
+// RingCfg is a wrapper for our Index Gateway ring configuration plus the replication factor.
+type RingCfg struct {
+	// InternalRingCfg configures the Index Gateway ring.
+	loki_util.RingConfig `yaml:",inline"`
+
+	// ReplicationFactor defines how many Index Gateway instances are assigned to each tenant.
+	//
+	// If the replication factor is higher than the number of instances, the ring is considered not ready.
```

Comment:
This line sounds incorrect because we will require a quorum of instances to respond with default replication strategy. 

This made me realize that while we are initializing ring with `NewIgnoreUnhealthyInstancesReplicationStrategy` for index gateway server, we are using default replication strategy for initializing ring for gateway client which would require a quorum of instances. We should switch to `NewIgnoreUnhealthyInstancesReplicationStrategy` for initializing  ring for index gateway client as well which would be happy even with 1 working index gateway instance.

Reply:
Pushed a commit addressing it, lmk what you think.